base_classes/NXlens_em.nxdl.xml

```diff
@@ -0,0 +1,110 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<?xml-stylesheet type="text/xsl" href="nxdlformat.xsl"?>
+<!--
+# NeXus - Neutron and X-ray Common Data Format
+# 
+# Copyright (C) 2014-2024 NeXus International Advisory Committee (NIAC)
+# 
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# For further information, see http://www.nexusformat.org
+-->
+<!--
+has_part pole_piece https://purls.helmholtz-metadaten.de/emg/EMG_00000039-->
+<!--
+more consolidation to harvest from a generic component base class
+with other research fields (MPES, XPS, OPT) could be useful-->
+<definition xmlns="http://definition.nexusformat.org/nxdl/3.1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" category="base" type="group" name="NXlens_em" extends="NXobject" xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd">
+    <doc>
+         Base class for an electro-magnetic lens or a compound lens.
+         
+         For :ref:`NXtransformations` the origin of the coordinate system is placed
+         in the center of the lens (its polepiece, pinhole, or another
+         point of reference). The origin should be specified in the :ref:`NXtransformations`.
+         
+         For details of electro-magnetic lenses in the literature
+         see e.g. `L. Reimer &lt;https://doi.org/10.1007/978-3-540-38967-5&gt;`_
+    </doc>
+    <!--user perspective-->
+    <field name="name" type="NX_CHAR">
+        <doc>
+            Name of the lens.
+        </doc>
+    </field>
+    <field name="description" type="NX_CHAR">
+        <doc>
+            Ideally, use instances of ``identifierNAME`` to point to a resource
+            that provides further details.
+
+            If such a resource does not exist or should not be used, use this free text,
+            although it is not recommended.
+        </doc>
+    </field>
+    <field name="power_setting" type="NX_NUMBER" units="NX_ANY">
```

Comment:
I have no experience with EM, but an observation as an outsider ...

I'm wondering whether this value is continuous (accepting a very large number of different values) or whether this value accepts only a modest number of values.

If it's the latter, then it might be more flexible to consider the values as arbitrary labels that just coincidentally look like numbers (i.e., `type="NX_CHAR"`).  This would allow the field to capture non-numerical values, should EM manufacturers start to use them.

Reply:
OK, thanks for the explanation.  I think we can leave this until there's an actual need for it.